libs/openFrameworks/gl/ofGLProgrammableRenderer.cpp

```diff
@@ -1315,12 +1321,46 @@ void ofGLProgrammableRenderer::bind(const ofFbo & fbo, bool setupPerspective){
 }
 
 //----------------------------------------------------------
-void ofGLProgrammableRenderer::unbind(const ofFbo & fbo){
-	glBindFramebuffer(GL_FRAMEBUFFER, 0);
+void ofGLProgrammableRenderer::end(const ofFbo & fbo){
+	fbo.unbind();
```

Comment:
Thanks so much for clearing this up- it makes all sense now =)

I'll see to send a fix tomorrow.

Btw would it be a good idea to somewhere formalise or externalise the design guidelines for the API and the renderer that were developed over time?

 I think it could be a helpful reminder for us and anyone wanting to send PRs...

I could start whacking something up into an issue or the GitHub wiki with the label "design strategy" which we could link to from the Contribute readme.

Some general principles around:
- which states the renderer tracks and which not
- where the renderer's responsibility ends 
- how GL state will be restored in core if in doubt
- gl best practices for the renderers